src/veins/modules/mobility/traci/subscriptionManagement/TraCITrafficLight.h

```diff
@@ -0,0 +1,81 @@
+//
+// Copyright (C) 2006-2017 Christoph Sommer <sommer@ccs-labs.org>
+//
+// Documentation for these modules is at http://veins.car2x.org/
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+//
+
+#ifndef SRC_VEINS_MODULES_MOBILITY_TRACI_SUBSCRIPTIONMANAGEMENT_TRACITRAFFICLIGHT_H_
+#define SRC_VEINS_MODULES_MOBILITY_TRACI_SUBSCRIPTIONMANAGEMENT_TRACITRAFFICLIGHT_H_
+
+#include <string>
+
+#include "veins/veins.h"
+
+namespace veins {
+namespace TraCISubscriptionManagement {
+
+/**
+ * @struct TraCITrafficLight
+ *
+ * Simple definition of a traffic light as data
+ * exchange format.
+ *
+ * @author Nico Dassler <dassler@hm.edu>
+ */
+struct TraCITrafficLight {
+    /**
+     * The id of the traffic light.
+     */
+    std::string id;
+
+    /**
+     * The index of the current phase in the current program.
+     */
+    int32_t currentPhase;
+
+    /**
+     * The id of the current program.
+     */
+    std::string currentProgram;
+
+    /**
+     * The assumed time (in seconds) at which the tls changes
```

Comment:
i can remove that but as stated in the comment this is just copied from the official documentation :P
--> https://sumo.dlr.de/wiki/TraCI/Traffic_Lights_Value_Retrieval

Reply:
True, but only since SUMO 1.0. We actually jumped through some loops to keep Veins compatible with older SUMO versions, where times in TraCI were still in milliseconds.
Also, time unit is abstracted away by `simtime_t`

Reply:
Removed.

Reply:
Great.